### PR TITLE
SOL-151649: Add LogValue, String, and GoString guards to Token, Excha…

### PR DESCRIPTION
**Pull request by solace-awaheed**
…ngeInput, Params, and DeduplicationKeyInput

## Summary

Adds String(), GoString(), and LogValue() guards to four credential-carrying types in internal/tokenexchange so their secret fields can never leak through slog reflection or fmt formatting.

Fixes # SOL-151649

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Dependency update

## Motivation and Context

slog.Any(...) on a struct that does not implement slog.LogValuer falls back to reflection and emits every exported field verbatim — including credential material. The same applies to fmt.Sprintf("%v"/"%+v", ...) when String()/GoString() are absent. A guard on an inner credential type is bypassed when the enclosing struct has no guard of its own, because slog and fmt only consult the outermost type. 

## Changes Made

  - Added String(), GoString(), and LogValue() (value receivers) to Token, ExchangeInput, Params, and DeduplicationKeyInput. Each omits the credential field and exposes only non-credential fields:
    - Token → expires_at
    - ExchangeInput → broker_alias, audience
    - Params → idp_endpoint, client_id, client_auth_method, grant_type, audience_param (infrastructure fields HTTPClient/Cache omitted)
    - DeduplicationKeyInput → broker_alias
  - Value receivers ensure *Token (returned by Exchange) and pointer forms of the other types are covered too.
  - Keyed the IdP URL in Params.LogValue as idp_endpoint rather than token_url: the ReplaceAttr net in cmd/server/main.go redacts any key containing "token", which would have silently blanked this non-credential field and misled operators.

## Testing

**Test Environment:**
- Go version:
- OS:
- Broker version (if applicable):

**Tests Performed:**
- [ ] Unit tests added/updated
- [ ] Integration tests added/updated (if applicable)
- [ ] E2E tests added/updated (if applicable)
- [x] Tested locally with `go test -race ./...`
- [ ] Tested with real Solace broker

**Test Coverage:**

  - Regression tests are explicitly out of scope for this story per SOL-151649 and are tracked in a separate ticket (slog.Any,
  fmt.Sprintf, and ExchangeError confirmation per type). Existing 142 package tests remain green.

## Breaking Changes

**Impact:**

**Migration Guide:**

### Code Quality
- [ ] Code follows the [coding standards](../.github/CONTRIBUTING.md#coding-standards)
- [ ] Self-review completed (checked for typos, logic errors, edge cases)
- [ ] Code is well-commented (explains "why", not "what")
- [ ] Complex logic has explanatory comments
- [ ] No commented-out code or debug statements left in

### Security
- [ ] **No credentials in code** (checked all files, commits, and logs)
- [ ] Followed [secure logging rules](../docs/secure-logging-rules.md)
- [ ] Credential-carrying types implement `slog.LogValuer`
- [ ] No security vulnerabilities introduced (checked with golangci-lint/gosec)

### Testing
- [ ] All tests pass locally: `go test -race ./...`
- [ ] No race conditions detected
- [ ] Edge cases covered by tests
- [ ] Error paths tested
- [ ] Test names clearly describe what is being tested

### Documentation
- [ ] README.md updated (if user-facing changes)
- [ ] docs/ updated (if architecture/design changes)
- [ ] godoc comments added/updated for exported functions
- [ ] CHANGELOG.md updated (if applicable)
- [ ] Examples updated (if applicable)

### Git & CI
- [ ] All commits are signed off (DCO: `git commit -s`)
- [ ] Commits have clear, descriptive messages
- [ ] Branch is up to date with main (rebased if needed)
- [ ] No merge conflicts
- [ ] CI checks passing (lint, build, test, E2E)

### Breaking Changes (if applicable)
- [ ] Breaking changes documented in commit message
- [ ] Breaking changes documented in PR description
- [ ] Migration guide provided
- [ ] Deprecated functionality marked with comments and deprecation notices

## Related Issues/PRs

- Related to #
- Depends on #
- Blocks #

---

**For Reviewers:**

**Focus Areas**: 

**Questions**:

### CODE REVIEW

**Review by solace-wmaryam — APPROVED** [review]
LGTM

File: internal/tokenexchange/types.go [hunk 1]
```diff
@@ -123,6 +125,33 @@ type Params struct {
 	Cache cache.TokenCache
 }
 
+// String, GoString, and LogValue redact ClientSecret so Params never leaks
+// the OAuth client secret through fmt formatting or slog reflection. They
+// expose only the non-credential protocol fields; the infrastructure fields
+// (HTTPClient, Cache) are omitted as noise. Value receivers so *Params is
+// covered too. Pattern mirrors cache.CachedCredential.
```
[the remark this answers is not in the document]
Comment (solace-awaheed) [reply internal/tokenexchange/types.go:132]: Testing not covered in this ticket.

File: internal/tokenexchange/dedup_key.go [hunk 2]
```diff
@@ -31,6 +33,22 @@ type DeduplicationKeyInput struct {
 	BrokerAlias  string
 }
 
+// String, GoString, and LogValue redact SubjectToken so
+// DeduplicationKeyInput never leaks it through fmt formatting or slog
+// reflection. Value receivers so *DeduplicationKeyInput is covered too.
+// Pattern mirrors cache.CachedCredential.
+func (d DeduplicationKeyInput) String() string {
```
[the remark this answers is not in the document]
Comment (solace-awaheed) [reply internal/tokenexchange/dedup_key.go:40]: Testing not covered in this ticket.
